### DISCUSSION TITLE
Keep it free for code contributors or beta users

### DISCUSSION THREAD

**Discussion by martin-braun** [community/community#19223] [Copilot Conversations] [answered]
I get that Copilot has a great opportunity to generate a lot of income, but I think you should allow anybody to use it for free as long as the programmer accepts that his code being fed into the AI which will improve the system even further.

It's an absolute low move to allow free access to anyone in the beta and now ask for money to continue to use it. The "worst, but acceptable" move would be to ask new users for money, but keep the current features for free for beta users, forever.

We beta users got used to it and now you rip it away from us. Instead improve the product and charge for the upgrade.

**Comment by vedantmgoyal9** [comment]
You can use it for free if you're a student. Just enroll for github student developer pack.

**Reply by martin-braun** [reply]
I'm not a student. This is a shameful move by Microsoft. They kicked beta users out from using their product without swiping their credit card. I'm really disappointed by this move, since I didn't know that it will cost in the end. I could pay it, but I think they should reward their beta testers with free access to the current features and only charge them for add-ons, like the upcoming Visual Studio extension. What they did is giving us a gift and then ripping it out of our hands after we got used to it, saying they want money for it from now on.

**Reply by Qamar255** [reply]
free mining bitcoin

**Comment by martin-braun** [comment, marked as answer]
Duplicate #19047

**Reply by Qamar255** [reply]
free mining bitcoin ?
